packages/components/button/src/use-button-config.ts

```diff
@@ -0,0 +1,22 @@
+import { provide, inject, getCurrentInstance, computed, shallowRef } from 'vue'
+import { ButtonGlobalConfigInjectionKey } from './button'
+import type { ButtonGlobalConfig } from './button'
+
+export function useButtonConfig() {
+  const vm = getCurrentInstance()!
+  const props = vm.props as {
+    button: ButtonGlobalConfig
+  }
+  const config = computed(() => props.button)
+  provide(ButtonGlobalConfigInjectionKey, config)
+}
+
+export function useButtonConfigInject() {
+  const config = inject(
+    ButtonGlobalConfigInjectionKey,
+    shallowRef({
+      autoInsertSpace: true,
+    })
```

Comment:
```suggestion
  return inject(ButtonGlobalConfigInjectionKey, undefined)
```
no need create a new function

Reply:
The default value is still required

Reply:
Refer to https://github.com/element-plus/element-plus/pull/4175/files#diff-1c64d60120c5ede235d1dcb98efa20a889756b54269c59b065eb7f27f5bf79d0L54 please